### PR TITLE
ui: add "total" values for disk and memory to node list; add num CPUs

### PR DESCRIPTION
**Pull request by vilterp**
image

There may be a more visually concise way of showing this.

### CODE REVIEW

**Comment by bdarnell** [comment]
Showing both the total capacity and the percentage is kind of redundant. I'd just show one or the other (probably the percentage) and leave the other for a mouseover to save a bit of horizontal space.

**Comment by vilterp** [comment]
Updated. It doesn't save a whole lot of horiz space because of the column headers, but it is less cluttered. TODO: use a real tooltip, not html `title`.

image

**Comment by robert-s-lee** [comment]
Could the same be done to CPU.  In the example, 4 cores (25%)

**Review by BramGruneir — CHANGES_REQUESTED** [review]
https://github.com/cockroachdb/cockroach/commit/1f64cec1b0374a7d513e65b6978615dad18641ee really needs more in the description.  There's a lot of code changes here and the summary in the commit message doesn't capture that.

What was the rationale behind the reordering in https://github.com/cockroachdb/cockroach/commit/cfb2ab336ed073a0eb66ebf120e4228c1a32c0cd.  Please add that to the commit message.

The final commit isn't needed.  It should be fixed back in the original commits for the changes.

Overall, the changes LGTM, but please go back and fixup the prior commits instead of issuing new ones. 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 10 of 10 files at r3, 1 of 1 files at r4, 4 of 4 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, 1 of 1 files at r8.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28189#-:-LJGDDZVAWbMjIl2Ja01:b-z1vuh4)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/server/status/recorder.go, line 604 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/28189#-LJGDZSd30Grw-cAEKK6:-LJGDZSd30Grw-cAEKK7:b-i9vq37) ([raw file](https://github.com/cockroachdb/cockroach/blob/1f64cec1b0374a7d513e65b6978615dad18641ee/pkg/server/status/recorder.go#L604)):*
> ```Go
> // GetTotalMemory returns either the total system memory or if possible the
> // cgroups available memory.
> func GetTotalMemory(ctx context.Context) (int64, error) {
> ```

Why did you move this?  There's no mention the commit message.
___
*[pkg/server/status/status.proto, line 61 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/28189#-LJGDtN6-fSAP7qD4Naw:-LJGDtN6-fSAP7qD4Nax:b-27nro6) ([raw file](https://github.com/cockroachdb/cockroach/blob/48536c3e29022fcaf409fa41e834e76c144140d2/pkg/server/status/status.proto#L61)):*
> ```protobuf
>   ];
>   // total_system_memory is the total RAM available to the system
>   // (or, if possible, the memory available to the cgroup this process is in)
> ```

This comment update should be on the commit where total_system_memory was added.

**Comment by vilterp** [comment]
TFTR — squashed down to a backend and a frontend commit, and added to the commit descriptions. PTAL.

**Review by BramGruneir — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r9, 2 of 2 files at r10.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28189#-:-LJKMK5Y61rZ93k-PufG:bnfp4nl)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

**Comment by vilterp** [comment]
TFTR!

bors r+

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 1]
```diff
@@ -6,16 +6,18 @@ import { createSelector } from "reselect";
...
 import { AdminUIState } from "src/redux/state";
 import { refreshNodes, refreshLiveness } from "src/redux/apiReducers";
 import { LocalSetting } from "src/redux/localsettings";
 import { SortSetting } from "src/views/shared/components/sortabletable";
 import { SortedTable } from "src/views/shared/components/sortedtable";
 import { LongToMoment } from "src/util/convert";
-import { Bytes } from "src/util/format";
+import { BytesWithPrecision } from "src/util/format";
 import { INodeStatus, MetricConstants, BytesUsed } from "src/util/proto";
+import {FixLong} from "oss/src/util/fixLong";
```
Comment (couchand) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: plz cleanup this line

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 2]
```diff
@@ -112,22 +114,47 @@ class LiveNodeList extends React.Component<NodeCategoryListProps, {}> {
...
             // Replicas - displays the total number of replicas on the node.
             {
               title: "Replicas",
               cell: (ns) => ns.metrics[MetricConstants.replicas].toString(),
               sort: (ns) => ns.metrics[MetricConstants.replicas],
             },
+            // CPUs - the number of CPUs on this node
+            {
+              title: "CPUs",
+              cell: (ns) => ns.num_cpus,
```
Comment (couchand) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: it seems weird to sort on num CPUs, but it's also a little weird that this column wouldn't be sortable...

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 3]
```diff
@@ -112,22 +114,47 @@ class LiveNodeList extends React.Component<NodeCategoryListProps, {}> {
...
+            },
+            // Used Capacity - displays the total persisted bytes maintained by the node.
+            {
+              title: "Capacity Usage",
+              cell: (ns) => {
+                const { usable } = nodeCapacityStats(ns);
+                const used = BytesUsed(ns);
```
Comment (couchand) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: I'm a bit uncomfortable that we're discarding `used` from `nodeCapacityStats` here in favor of calling `BytesUsed`, but elsewhere we just use that.  let's make these consistent, please

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 4]
```diff
@@ -112,22 +114,47 @@ class LiveNodeList extends React.Component<NodeCategoryListProps, {}> {
...
+                return (
+                  <span title={`Total: ${BytesWithPrecision(usable, 0)}`}>
+                    {BytesWithPrecision(used, 0)}
```
Comment (couchand) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: why are we throwing away the tenths place?  the whole point of using `Bytes` everywhere is that we're consistent in our reporting across the whole project.

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 5]
```diff
@@ -112,22 +114,47 @@ class LiveNodeList extends React.Component<NodeCategoryListProps, {}> {
...
+                    {" "}
+                    ({Math.round(used / usable * 100)}%)
```
Comment (couchand) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: how about `Percentage` from `util/format` instead?

File: pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx [hunk 6]
```diff
@@ -112,22 +114,47 @@ class LiveNodeList extends React.Component<NodeCategoryListProps, {}> {
...
+              },
+              sort: (ns) => BytesUsed(ns),
+            },
             // Mem Usage - total memory being used on this node.
             {
               title: "Mem Usage",
-              cell: (ns) => Bytes(ns.metrics[MetricConstants.rss]),
+              cell: (ns) => {
+                const used = ns.metrics[MetricConstants.rss];
+                const available = FixLong(ns.total_system_memory).toNumber();
```
Comment (couchand) [inline pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx]: i'd prefer if we did similar treatment in `nodeSumStats` for memory as we do for capacity, to keep things consistent

**Pull request merged**